iterative/utils/logger.go

```diff
@@ -0,0 +1,104 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/sirupsen/logrus"
+	"strings"
+	"time"
+)
+
+var baseTimestamp = time.Now()
+var colors = make(map[string]int)
+
+type basicFormatter struct{}
+
+func (f *basicFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	levelText := strings.ToUpper(entry.Level.String())
+	levelColor := colors[levelText]
+	tpl := "[%s] 🚀\x1b[%dmTPI\x1b[0m %s\n"
+	return []byte(fmt.Sprintf(tpl, levelText, levelColor, entry.Message)), nil
+}
+
+func init() {
+	colors["DEBUG"] = 36
+	colors["INFO"] = 36
+	colors["WARN"] = 33
+	colors["ERROR"] = 31
+	colors["FATAL"] = 31
+	colors["purple"] = 35
+
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.SetFormatter(&basicFormatter{})
+}
+
+type tpiFormatter struct{}
+
+func (f *tpiFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	data := make(logrus.Fields)
+	for k, v := range entry.Data {
+		data[k] = v
+	}
+
+	d := data["d"].(*schema.ResourceData)
+	message := entry.Message
+	levelText := strings.ToUpper(entry.Level.String())
+	levelColor := colors[levelText]
+
+	if message == "instance" {
+		cloud := d.Get("cloud").(string)
+		machine := d.Get("machine").(string)
+		region := d.Get("region").(string)
+		spot := d.Get("spot").(float64)
+
+		spottext := ""
+		if spot > 0 {
+			spottext = fmt.Sprintf("(Spot %f/h)", spot)
+		}
+		message = fmt.Sprintf("🚀 %s %s%s at %s", cloud, machine, spottext, region)
```

Comment:
The :rocket: is already in the template, right?

```suggestion
		message = fmt.Sprintf("%s %s%s in %s", cloud, machine, spottext, region)
```